SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt

```diff
@@ -21,18 +33,61 @@ class ApplicationPresenter: ApplicationContract.Presenter() {
 
     private var departureStation: Station? = null
     private var arrivalStation: Station? = null
+    private var model: FaresModel? = null
 
     override val coroutineContext: CoroutineContext
         get() = dispatchers.main + job
 
     override fun onViewTaken(view: ApplicationContract.View) {
-        this._view = view
+        this.view = view
         view.setLabel(createApplicationScreenMessage())
-        view.setStations(stations);
+        view.setStations(stations)
     }
 
     override fun onTimesRequested() {
-        val now = "2020-10-14T19:30:00.000+01:00"
+        val view = view
```

Comment:
This line doesn't appear to be doing anything useful?

Reply:
It's making the view locally static so it can be used within the coroutine scope.

Reply:
I don't think you actually need to do this, though I'll be honest I'm currently struggling to find better justification for that than "it still works if you don't" - you're already using `model` within the coroutine which is defined on the class

Reply:
oh you're right - we were doing the coroutine stuff a different (outdated) way at first using GlobalScope.launch which would error if we didn't do this - didn't realise it worked now.